doc: peripherals: unify title strings

henrikbrixandersen opened the pull request:
Unify the peripheral documentation title strings to the format "<class> [(acronym)] [Bus]".

Including both the full name of the peripheral class and an acronym makes the documentation more user friendly as some of the acronyms are less well-known than others.

This is a follow-up to #56275.

henrikbrixandersen committed:
doc: peripherals: unify title strings

Unify the peripheral documentation title strings to the format
"<class> [(acronym)] [Bus]".

Including both the full name of the peripheral class and an acronym makes
the documentation more user friendly as some of the acronyms are less
well-known than others.

Signed-off-by: Henrik Brix Andersen <hebad@vestas.com>

martinjaeger reviewed (dismissed):
Typo in the commit message (peri**hp**erals). Otherwise LGTM. Thanks!

henrikbrixandersen commented:
> Typo in the commit message (peri**hp**erals). Otherwise LGTM. Thanks!

Fixed, thank you.

anangl commented on doc/hardware/peripherals/adc.rst:
```diff
@@ -1,7 +1,7 @@
 
-ADC
-###
+Analog-to-Digial Converter (ADC)
```
```suggestion
Analog-to-Digital Converter (ADC)
```

henrikbrixandersen replied:
Ouch, thanks. Fixed.

anangl commented on doc/hardware/peripherals/espi.rst:
```diff
@@ -1,7 +1,7 @@
 
-eSPI
-####
+Enhanced Serial Peripheral Interface (eSPI) bus
```
```suggestion
Enhanced Serial Peripheral Interface (eSPI) Bus
```

henrikbrixandersen replied:
Fixed.

kartben reviewed (changes requested):
This is very cool, thank you for proposing this change!

I would suggest the following additions:

- Gaussian & Neural Accelerator (GNA)
- Real-Time Clock (RTC)
- USB Type-C Port Controller (TCPC)
- MIPI Display Serial Interface (DSI)

Also, could you make sure index.rst remains (or rather, is) alphabetically ordered (e.g 1-Wire should now be showing up first).

Thank you!

henrikbrixandersen committed:
doc: peripherals: sort ToC according to subpage titles

Sort the peripherals table of contents according to the titles of the pages
included.

Signed-off-by: Henrik Brix Andersen <hebad@vestas.com>

kartben reviewed (approved):
Thank you!
